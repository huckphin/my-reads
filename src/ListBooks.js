import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BookShelf from './BookShelf';

class ListBooks extends Component {
    static propTypes = {
        books: PropTypes.array.isRequired
    }

    render () {
        const { books } = this.props;

        const currentlyReading = books.filter((book) => book.shelf === 'currentlyReading');
        const read = books.filter((book) => book.shelf === 'read');
        const wantToRead = books.filter((book) => book.shelf === 'wantToRead');

        console.log(currentlyReading);

        return (
            <div className="list-books">
                <div className="list-books-title">
                    <h1>MyReads</h1>
                </div>
                <div className="list-books-content">
                    <div>
                        <BookShelf
                            title="Currently Reading"
                            key="Currently Reading"
                            books={currentlyReading}
                        />
                        <BookShelf
                            title="Want to Read"
                            key="Want to Read"
                            books={[
                                {
                                    authors: [ "David McCullugh" ],
                                    imageLinks: {
                                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                                    },
                                    title: "1776",
                                    key: "1776"
                                },
                                {
                                    authors: [ "J.K. Rowling" ],
                                    imageLinks: {
                                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                                    },
                                    title: "Harry Potter and the Sorcerer's Stone",
                                    key: "Harry Potter and the Sorcerer's Stone"
                                }
                            ]}
                        />
                        <BookShelf
                            title="Read"
                            key="Read"
                            books={[
                                {
                                    authors: [ "J.R.R. Tolkien" ],
                                    imageLinks: {
                                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                                    },
                                    title: "The Hobbit",
                                    key: "The Hobbit"
                                },
                                {
                                    authors: [ "Seuss" ],
                                    imageLinks: {
                                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                                    },
                                    title: "Oh, the Places You'll Go!",
                                    key: "Oh, the Places You'll Go!"
                                },
                                {
                                    authors: [ "Mark Twain" ],
                                    imageLinks: {
                                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                                    },
                                    title: "The Adventures of Tom Sawyer",
                                    key: "The Adventures of Tom Sawyer"
                                }
                            ]}
                        />
                    </div>
                </div> {/* list-books-content */}
                <div className="open-search">
                    <a onClick={() => this.setState({ showSearchPage: true })}>Add a book</a>
                </div>
            </div>
        )
    }
}

export default ListBooks;