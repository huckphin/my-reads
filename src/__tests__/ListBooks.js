import React from 'react';
import ListBooks from '../ListBooks';
import renderer from 'react-test-renderer';

test('will render ListBooks', () => {
    const component = renderer.create(
        <ListBooks
            key="MyReads"
            books={[
                {
                    key: "The Linux Command Line",
                    authors: [ "William E. Shotts, Jr." ],
                    imageLinks: {
                        smallThumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                        thumbnail: "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
                    },
                    title: "The Linux Command Line",
                    shelf: "Currently Reading"
                }
            ]}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});