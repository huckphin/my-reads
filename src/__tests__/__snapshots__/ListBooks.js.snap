// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will render ListBooks 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Currently Reading
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          />
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Want to Read
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select>
                      <option
                        disabled={true}
                        value="none"
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-authors"
                >
                  David McCullugh
                </div>
                <div
                  className="book-title"
                >
                  1776
                </div>
              </div>
            </li>
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select>
                      <option
                        disabled={true}
                        value="none"
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-authors"
                >
                  J.K. Rowling
                </div>
                <div
                  className="book-title"
                >
                  Harry Potter and the Sorcerer's Stone
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
      <div
        className="bookshelf"
      >
        <h2
          className="bookshelf-title"
        >
          Read
        </h2>
        <div
          className="bookshelf-books"
        >
          <ol
            className="books-grid"
          >
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select>
                      <option
                        disabled={true}
                        value="none"
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-authors"
                >
                  J.R.R. Tolkien
                </div>
                <div
                  className="book-title"
                >
                  The Hobbit
                </div>
              </div>
            </li>
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select>
                      <option
                        disabled={true}
                        value="none"
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-authors"
                >
                  Seuss
                </div>
                <div
                  className="book-title"
                >
                  Oh, the Places You'll Go!
                </div>
              </div>
            </li>
            <li>
              <div
                className="book"
              >
                <div
                  className="book-top"
                >
                  <div
                    className="book-cover"
                    style={
                      Object {
                        "backgroundImage": "url(http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api)",
                      }
                    }
                  />
                  <div
                    className="book-shelf-changer"
                  >
                    <select>
                      <option
                        disabled={true}
                        value="none"
                      >
                        Move to...
                      </option>
                      <option
                        value="currentlyReading"
                      >
                        Currently Reading
                      </option>
                      <option
                        value="wantToRead"
                      >
                        Want to Read
                      </option>
                      <option
                        value="read"
                      >
                        Read
                      </option>
                      <option
                        value="none"
                      >
                        None
                      </option>
                    </select>
                  </div>
                </div>
                <div
                  className="book-authors"
                >
                  Mark Twain
                </div>
                <div
                  className="book-title"
                >
                  The Adventures of Tom Sawyer
                </div>
              </div>
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
   
  <div
    className="open-search"
  >
    <a
      onClick={[Function]}
    >
      Add a book
    </a>
  </div>
</div>
`;
